ED/src/init/ed_params.f90

```diff
@@ -4570,1076 +7248,1598 @@ end subroutine init_hydro_coms
+   print_zero_table = btest(idetailed,5)
+   !---------------------------------------------------------------------------------------!
+
 
+   !----- 25 degC, to find Vcmax25. -------------------------------------------------------!
+   temp25C8 = 2.5d1 + t008
+   !---------------------------------------------------------------------------------------!
 
 
 
+   !------ Make sure the soil carbon fractions add up to one. -----------------------------!
+   if (f0_msc < 0. .or. f0_psc < 0.0 .or. (f0_msc + f0_psc) > 1.0) then
+      write (unit=*,fmt='(a)')          '-------------------------------------------------'
+      write (unit=*,fmt='(a)')          ' F0_MSC and F0_SSC must be fractions (0-1)'
+      write (unit=*,fmt='(a)')          '    and their sum cannot exceed 1.0'
+      write (unit=*,fmt='(a)')          ''
+      write (unit=*,fmt='(a)')          ' Current values: '
+      write (unit=*,fmt='(a,1x,f12.5)') ' F0_MSC = ',f0_msc
+      write (unit=*,fmt='(a,1x,f12.5)') ' F0_PSC = ',f0_psc
+      write (unit=*,fmt='(a)')          '-------------------------------------------------'
+      write (unit=*,fmt='(a)')          ''
+      call fatal_error('Invalid partition of soil C pools. Fix ed_params.f90 or XML.'      &
+                      ,'init_derived_params_after_xml','ed_params.f90')
+   end if
+   f0_ssc = 1.0 - f0_msc - f0_psc
    !---------------------------------------------------------------------------------------!
-   !     Fill in the albedo information regarding the soil colour classes.                 !
+
+
    !---------------------------------------------------------------------------------------!
-   !                    |    Dry soil   |   Saturated   | Emis. |                          !
-   !   Soil class       |---------------+---------------|-------|                          !
-   !                    |   VIS |   NIR |   VIS |   NIR |   TIR |                          !
+   !     Find net specific heat for leaves, sapwood, heartwood, and bark.  Here we modify  !
+   ! the calculation from earlier versions of ED-2.2.  We now separate oven-dry biomass    !
+   ! and internal water for leaves and sapwood, as the latter is dynamic.  For now heart-  !
+   ! wood and bark are assumed to have constant internal water content, so we incorporate  !
+   ! the specific heat of internal water to the bulk specific heat.                        !
+   !     In addition, we ignore the wood-water bonding heat capacity for sapwood.  This is !
+   !  a relative small effect (2-5% of heat capacity), and it only changes when water      !
+   !  content is very low.  Keeping this term may add some non-linearities and would       !
+   ! require dynamic updates --- not a big deal, and we may add this effect in future      !
+   ! versions.                                                                             !
+   !                                
```

Comment:
@serbinsh @ashiklom  starting here, could you review that the changes @mpaiao made reflect things that should be calculated from other ED optics variables and should NOT be set by the XML. We should make sure this is true for both the default code and RTM stand alone.

Reply:
I'll try to take a look at this later this week.